Ziti.NET.Standard/src/OpenZiti/ZitiIdentity.cs

```diff
@@ -467,44 +466,72 @@ public class ZitiRouterEvent {
 	}
 
 	public class ZitiServiceEvent {
-		internal IntPtr nativeRemovedList;
-		internal IntPtr nativeChangedList;
-		internal IntPtr nativeAddedList;
-		internal IntPtr ziti_ctx;
+
+		public ZitiServiceEvent(IntPtr zitiCtx) {
+			this.ziti_ctx = zitiCtx;
+		}
+		internal IntPtr nativeRemovedList {
+			set {
+				removedList = new List<ZitiService>();

```

Comment:
If the ZitiService objects are created inside the Added, Removed and Changed functions, it creates ZitiSevice objects for each native service, multiple times based on the number of times the enumerable object is called.